### ISSUE TITLE
dev-python/pyinstaller is missing dependencies

### ISSUE THREAD

**Issue by ikelos** [pentoo/pentoo-overlay#459]
Hiya,

On running pyinstaller it seems to use setuptools to check for dependencies (altgraph, macholib>=1.8 and pefile>=2017.8.1), which fail:

```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.6/pyinstaller", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3241, in <module>
    @_call_aside
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3225, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3254, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'altgraph' distribution was not found and is required by PyInstaller
```

```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.6/pyinstaller", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3241, in <module>
    @_call_aside
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3225, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3254, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'macholib>=1.8' distribution was not found and is required by PyInstaller
```

```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.6/pyinstaller", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3241, in <module>
    @_call_aside
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3225, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 3254, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pefile>=2017.8.1' distribution was not found and is required by PyInstaller
```

That therefore means the comment in the ebuild:
```
#FIXME: Interestingly, PyInstaller itself has no hard or soft dependencies
```
is a bit misleading.  Also, pyinstaller runs just fine under python3.7...  5;)

**Comment by linxon** [comment]
do you know what's going on here: [pyinstaller/bootloader/waf#L168](https://github.com/pyinstaller/pyinstaller/blob/ad3d2d46804d2d955c17eae740b8c2eac529611e/bootloader/waf#L168) ?

**Comment by ikelos** [comment]
I assume that's a packed archive which the waf unpacks.  I don't know much about waf I'm afraid, but according to https://en.wikipedia.org/wiki/Waf, it's the source for the waf system which then does the installation...

**Comment by linxon** [comment]
thank you! in the current moment i have one problem while using a `leak-detector` flag

More: https://github.com/pyinstaller/pyinstaller/issues/4196

**Comment by blshkv** [comment]
we copied the ebuild form glicOne  overlay. So all bugs belong to them ;-)

**Comment by blshkv** [comment]
fixed in d4d6e3f90b45832b71d6d6035ca8c0330e527245
